uaclient/api/u/pro/subscription/v1.py

```diff
@@ -0,0 +1,265 @@
+        self.origin = origin
+        self.effective = effective
+        self.expires = expires
+
+
+class AccountInfo(DataObject, AdditionalInfo):
+    fields = [
+        Field(
+            "created_at", DatetimeDataValue, doc="Creation date of the account"
+        ),
+        Field(
+            "external_account_ids",
+            data_list(StringDataValue),
+            doc="External account identifiers",
+        ),
+        Field("id", StringDataValue, doc="Account id"),
+        Field("name", StringDataValue, doc="Account name"),
+    ]
+
+    def __init__(
+        self,
+        *,
+        created_at: str,
+        external_account_ids: List[str],
+        id: str,
+        name: str,
+    ):
+        self.created_at = created_at
+        self.external_account_ids = external_account_ids
+        self.id = id
+        self.name = name
+
+
+class SubscriptionResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("contract", ContractInfo, doc="Contract Information"),
+        Field("account", AccountInfo, doc="Account information"),
+        Field("machine_id", StringDataValue, doc="Machine id"),
+        Field("activity_id", StringDataValue, doc="Activity id"),
+        Field(
+            "machine_is_attached",
+            BoolDataValue,
+            doc="Check if machine is attached to a contract",
+        ),
+    ]
+
+    def __init__(
+        self,
+        *,
+        contract: ContractInfo,
+        account: AccountInfo,
+        machine_id: str,
+        activity_id: str,
+        machine_is_attached: bool
+    ):
+        self.contract = contract
+        self.account = account
+        self.machine_id = machine_id
+        self.activity_id = activity_id
+        self.machine_is_attached = machine_is_attached
+
+
+def subscription() -> SubscriptionResult:
+    return _subscription(UAConfig())
+
+
+def _subscription(cfg: UAConfig) -> SubscriptionResult:
+    """
+    Returns the Ubuntu Pro subscription information for the machine.
+    """
+    if not _is_attached(cfg).is_attached:
+        return SubscriptionResult(
+            contract=ContractInfo(
+                created_at="",
+                id="",
+                name="",
+                products=[],
+                tech_support_level="",
+                origin=None,
+                effective=None,
+                expires=None,
+            ),
+            account=AccountInfo(
+                created_at="",
+                external_account_ids=[],
+                id="",
+                name="",
+            ),
+            machine_id="",
+            activity_id="",
+            machine_is_attached=False,
+        )
+
+    machine_token_file = machine_token.get_machine_token_file(cfg)
+    machineTokenInfo = machine_token_file.machine_token["machineTokenInfo"]
+
+    # Contract Info
+    contractInfo = machineTokenInfo["contractInfo"]
+    tech_support_level = UserFacingStatus.INAPPLICABLE.value
```

Comment:
We shouldn't use any `UserFacingStatus` values in the api results.
If there is not tech support level in the machine token info, then we should return null